### ISSUE TITLE
A Floating Action Button (FAB)

### ISSUE THREAD

**Issue by xenoterracide** [jgthms/bulma#2843]
Im not looking for a Material Design framework, however the [FAB](https://material.io/components/buttons-floating-action-button/#)  is the one thing I would like from that world. It does't appear to be available https://bulma.io/documentation/elements/button/ . Could we get one?

**Comment by rbrtbrnschn** [comment]
### agreed

**Comment by ravilution** [comment]
This is how I did. I needed my FAB to be in the bottom center. If you don't want it in center you can add `is-fab` class without `width:100%` directly to button.

```css
.is-fab {
  width: 100%;
  position: fixed;
  bottom: 0;
}
.has-content-centered {
  display: flex;
  justify-content: center;
}
```

```html
<div class="is-fab has-content-centered">
  <button class="button is-primary is-rounded">
    <span class="icon">
      <i class="fas fa-pen-fancy"></i>
    </span>
    <span>Post</span>
  </button>
</div>
```

**Comment by xenoterracide** [comment]
that code didn't work for me at all... the icon was off the screen, wasn't until I deleted all the css that it showed up.

**Comment by ravilution** [comment]
> that code didn't work for me at all... the icon was off the screen, wasn't until I deleted all the css that it showed up.

You will have to keep the `is-fab` `div` directly under `body`. I mean at the root of `body`.

**Comment by xenoterracide** [comment]
:/ that's not really going to work for my frameworked code

this seems to

```scss
.is-fab {
  position: fixed;
  right: 2rem;
  bottom: 2rem;
  :hover {
    color: bulmaDarken($primary, 2.5%);
  }
}
```html
<div class="is-fab">
    <span class="icon is-large has-text-primary">
        <i class="fas fa-plus-circle fa-4x"></i>
    </span>
</div>
```

**Comment by alakise** [comment]
Hello, 
I've made an extension for Bulma to create floating buttons [https://www.npmjs.com/package/bulma-floating-button](https://www.npmjs.com/package/bulma-floating-button)
Few days ago I wanted to use a floating button in my Bulma project too, then found this issue. Eventually, I could not find anything so I decided to create on my own. I hope it helps. I will create a pull request to Bulma soon.
![example](https://user-images.githubusercontent.com/17834259/81270141-f2983200-9052-11ea-98fe-0d8a2ef59f0f.png)
